docs/source/create_an_experiment_(development).rst

```diff
@@ -0,0 +1,723 @@
+info will store the recalled text. If the info creation succeeds, the
+`create_agent` function will be called:
+
+.. code-block:: javascript
+
+  var create_agent = function() {
+    $('#finish-reading').prop('disabled', true);
+    dallinger.createAgent()
+    .done(function (resp) {
+      $('#finish-reading').prop('disabled', false);
+      my_node_id = resp.node.id;
+      get_info();
+    })
+    .fail(function (rejection) {
+      if (rejection.status === 403) {
+        dallinger.allowExit();
+        dallinger.goToPage('questionnaire');
+      } else {
+        dallinger.error(rejection);
+      }
+    });
+  };
+
+The `create_agent` function is called twice in this experiment. The first
+time when the experiment page loads, and the second time when the
+`submit-response` button is clicked.
+
+Both times, it first disables the `finish-reading` button before calling the
+`dallinger.createAgent` function. This function calls the Python backend,
+to create an experiment node for the current participant.
+
+The first time, this call will succeed, since there is no node defined for
+this participant. In that case, we enable the `finish-reading` button and
+save the returned node's id in the `my_node_id` global variable defined at
+the start of our Javascript code. Finally, we call the `get_info` function
+defined below.
+
+The second time that `create_agent` is called, is when the text is
+submitted by the user. When that happens, the underlying `createAgent` call
+will fail, and return a rejection status of "403". The code above checks
+for that status, and if it finds it, that's the signal for us to finish
+the experiment and send the user to the Dallinger questionnaire page. If
+the rejection status is not "403", that means something unexpected
+happened, and we need to raise a Dallinger error, effectively ending the
+experiment.
+
+Now let's discuss the `get_info` function mentioned above, which is
+called when the experiment first calls the `create_agent` function:
+
+.. code-block:: javascript
+
+  var get_info = function() {
+    dallinger.getReceivedInfos(my_node_id)
+    .done(function (resp) {
+      var story = resp.infos[0].contents;
+      $("#story").html(story);
+      $("#stimulus").show();
+      $("#response-form").hide();
+      $("#finish-reading").show();
+    })
+    .fail(function (rejection) {
+      console.log(rejection);
+      $('body').html(rejection.html);
+    });
+  };
+
+Remember that in the Python code above, in the `add_node_to_network`
+method, we looked for the participant's parent, and then called its
+`transmit` method, followed by the node's own `receive` method. This
+transmits the parent node's info to the new node. The Javascript `get_info`
+function tries to get that info by calling `dallinger.getReceivedInfos` with
+the node id that we saved after successfully calling `dallinger.createAgent`.
+
+For the first participant, this info will contain the text generated by the
```

Comment:
It's worth emphasising that this is because it's a chain. The parent of the first participant is the original source. The second participant's parent is the first participant, so it transmits the result of the `createInfo` call.